[dtensor] add debug tool to track op coverage

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100124

This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

wanchaol committed:
[dtensor] add debug tool to track op coverage

This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

[ghstack-poisoned]

XilunWu commented on torch/distributed/_tensor/debug/op_coverage.py:
```diff
@@ -0,0 +1,38 @@
+
+def get_inductor_decomp_graphs(model: nn.Module, args, kwargs):
+    """
+    Convinient util to get the fwd and bwd graphs of an arbitrary model
```
typo: convinient -> convenient

XilunWu commented on test/distributed/_tensor/debug/test_op_coverage.py:
```diff
@@ -0,0 +1,34 @@
+        kwargs = {}
+        graphs = get_inductor_decomp_graphs(model, args, kwargs)
+        assert len(graphs) == 2, "Expect fwd + bwd graphs"
+        self.assertIsInstance(graphs[0], torch.fx.GraphModule)
```
question: will we check the content of the graphs in this test or another? Does this PR include checking the graph content or just validating that the util can return 1 fwd and 1 bwd graphs.

wanchaol replied:
This PR would not check the content of the graph in the tests, I added a `print_op_summary` function that would print the op coverage summary

XilunWu reviewed (approved):
LGTM! a good start for op coverage!!

wanchaol committed:
Update on "[dtensor] add debug tool to track op coverage"


This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] add debug tool to track op coverage"


This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] add debug tool to track op coverage"


This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] add debug tool to track op coverage"


This PR adds a debug tool to track the op coverage needed in DTensor.

Note that we specifically target ops after decomp table in inductor

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
